common/persistence/nosql/nosqlplugin/cassandra/workflow_utils_test.go

```diff
@@ -0,0 +1,858 @@
+				CurrentWorkflowConditionFailInfo: common.StringPtr(
+					"Workflow execution creation condition failed. WorkflowId: wfid, " +
+						"CurrentRunID: bda9cd9c-32fb-4267-b120-346e5351fc46"),
+			},
+		},
+		{
+			desc: "execution already running",
+			session: &fakeSession{
+				mapExecuteBatchCASApplied: false,
+				iter:                      &fakeIter{},
+				mapExecuteBatchCASPrev: map[string]any{
+					"type":                        rowTypeExecution,
+					"run_id":                      uuid.Parse("bda9cd9c-32fb-4267-b120-346e5351fc46"),
+					"range_id":                    int64(100),
+					"workflow_last_write_version": int64(3),
+				},
+			},
+			batch:        &fakeBatch{},
+			currentWFReq: &nosqlplugin.CurrentWorkflowWriteRequest{},
+			execution: &nosqlplugin.WorkflowExecutionRequest{
+				InternalWorkflowExecutionInfo: persistence.InternalWorkflowExecutionInfo{
+					WorkflowID:      "wfid",
+					RunID:           "bda9cd9c-32fb-4267-b120-346e5351fc46",
+					CreateRequestID: "reqid_123",
+					State:           2,
+				},
+			},
+			shardCond: &nosqlplugin.ShardCondition{
+				RangeID: 100,
+			},
+			wantErr: &nosqlplugin.WorkflowOperationConditionFailure{
+				WorkflowExecutionAlreadyExists: &nosqlplugin.WorkflowExecutionAlreadyExists{
+					OtherInfo: "Workflow execution already running. WorkflowId: wfid, " +
+						"RunId: bda9cd9c-32fb-4267-b120-346e5351fc46, rangeID: 100",
+					CreateRequestID:  "reqid_123",
+					RunID:            "bda9cd9c-32fb-4267-b120-346e5351fc46",
+					State:            2,
+					LastWriteVersion: 3,
+				},
+			},
+		},
+		{
+			desc: "unknown condition failure",
+			session: &fakeSession{
+				mapExecuteBatchCASApplied: false,
+				iter:                      &fakeIter{},
+				mapExecuteBatchCASPrev: map[string]any{
+					"type":                        rowTypeExecution,
+					"run_id":                      uuid.Parse("bda9cd9c-32fb-4267-b120-346e5351fc46"),
+					"range_id":                    int64(100),
+					"workflow_last_write_version": int64(3),
+				},
+			},
+			batch:        &fakeBatch{},
+			currentWFReq: &nosqlplugin.CurrentWorkflowWriteRequest{},
+			execution: &nosqlplugin.WorkflowExecutionRequest{
+				InternalWorkflowExecutionInfo: persistence.InternalWorkflowExecutionInfo{
+					RunID: "something else",
+				},
+			},
+			shardCond: &nosqlplugin.ShardCondition{
+				RangeID: 100,
+			},
+			wantErr: &nosqlplugin.WorkflowOperationConditionFailure{
+				UnknownConditionFailureDetails: common.StringPtr("Failed to operate on workflow execution.  Request RangeID: 100"),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.desc, func(t *testing.T) {
+			err := executeCreateWorkflowBatchTransaction(tc.session, tc.batch, tc.currentWFReq, tc.execution, tc.shardCond)
+			if diff := errDiff(tc.wantErr, err); diff != "" {
+				t.Fatalf("Error mismatch (-want +got):\n%s", diff)
+			}
+			if err != nil {
+				return
+			}
+			if !tc.session.iter.closed {
```

Comment:
should we move this check before the return statement?

Reply:
will address in part-2